035-rebalance-types-scaling-brokers.md

```diff
@@ -0,0 +1,117 @@
+# Extending the `KafkaRebalance` resource with rebalance types to help scaling the Apache Kafka cluster
+
+This proposal is about leveraging the Cruise Control integration in the Strimzi cluster operator in order to run an Apache Kafka cluster rebalancing when it's scaled up or down.
+It's related to add the integration with two specific Cruise Control API endpoints for helping to move replicas to newly added brokers or move them out from removed ones, during the scaling operations.
+The proposal doesn't assume any automation to run the rebalancing, it is always user driven by using the improved `KafkaRebalance` custom resource. See the [Flow](#flow) section for more details.
+It has to be considered as "phase 1" towards more automation.
+The future plan is about having the user able to specify some criteria to match in order to have the Strimzi cluster operator able to run the rebalancing operation automatically during the scale up and down, reducing the user manual intervention.
+
+## Current situation
+
+Currently, when Cruise Control is deployed alongside the Apache Kafka cluster, no automatic rebalancing happens when the cluster itself is scaled up or down.
+
+It means that, if the cluster is scaled up, no replicas will be moved to the newly added brokers.
+These brokers will only get partition replicas of newly created topics.
+Furthermore, by using the `KafkaRebalance` custom resource to run a rebalancing operation after the scaling up, it will move replicas across all the already existing brokers and not only to the new ones.
+It actually runs a full-rebalance with the drawback of more data movement, CPU usage and so on.
+
+In the same way, if the cluster is scaled down, the replicas hosted on the target brokers to remove are not moved out before the scaling down happens.
+Right now the scaling down is not blocked but doing it is not part of this proposal.
```

Comment:
Have we got an issue for this? I think @ShubhamRwt said he was planning to add this check?

Reply:
Yeah @ShubhamRwt is working on it but we didn't have an actual issue :-(
I created one https://github.com/strimzi/strimzi-kafka-operator/issues/6736 and now adding the link to the proposal.